### ISSUE TITLE
Unable to map KCor images from ucar.edu

### ISSUE THREAD

**Issue by dgarciabriseno** [sunpy/sunpy#6111]
### Describe the bug

When trying to create a map using one of the jp2 files from here: http://download.hao.ucar.edu/jp2/kcor/2022/04/29/white-light-pB/ sunpy fails to create the map.

### To Reproduce

Download http://download.hao.ucar.edu/jp2/kcor/2022/04/29/white-light-pB/2022_04_29__01_47_44__MLSO_KCOR_KCOR_white-light-pB.jp2

```python
>>> import sunpy.map
>>> sunpy.map.Map('2022_04_29__01_47_44__MLSO_KCOR_KCOR_white-light-pB.jp2')
```

### What happened?

```python
>>> sunpy.map.Map('2022_04_29__01_47_44__MLSO_KCOR_KCOR_white-light-pB.jp2')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/data/jp2/KCor/2022/04/29/735/venv/lib64/python3.8/site-packages/sunpy/map/map_factory.py", line 331, in __call__
    new_map = self._check_registered_widgets(data, meta, **kwargs)
  File "/data/jp2/KCor/2022/04/29/735/venv/lib64/python3.8/site-packages/sunpy/map/map_factory.py", line 381, in _check_registered_widgets
    return WidgetType(data, meta, **kwargs)
  File "/data/jp2/KCor/2022/04/29/735/venv/lib64/python3.8/site-packages/sunpy/map/sources/mlso.py", line 37, in __init__
    self.plot_settings['cmap'] = self._get_cmap_name()
  File "/data/jp2/KCor/2022/04/29/735/venv/lib64/python3.8/site-packages/sunpy/map/sources/mlso.py", line 46, in _get_cmap_name
    cmap_string = self.meta['detector']
  File "/data/jp2/KCor/2022/04/29/735/venv/lib64/python3.8/site-packages/sunpy/util/metadata.py", line 161, in __getitem__
    return OrderedDict.__getitem__(self, key.lower())
KeyError: 'detector'
```

### Expected behavior

Expect the map object to be returned.

### System Details

```python
>>> sunpy.util.system_info()
==============================
sunpy Installation Information
==============================

General
#######
OS: Oracle Linux Server (8.5, Linux 5.4.17-2136.302.7.2.el8uek.x86_64)
Arch: 64bit, (x86_64)
sunpy: 3.1.6
Installation path: /data/jp2/KCor/2022/04/29/735/venv/lib/python3.8/site-packages

Required Dependencies
#####################
astropy: 5.0.4
parfive: 1.5.1
packaging: 21.3
numpy: 1.22.3

Optional Dependencies
#####################
asdf: Missing, need asdf>=2.6.0; extra == "all"
beautifulsoup4: Missing, need beautifulsoup4>=4.8.0; extra == "all"
cdflib: Missing, need cdflib!=0.4.0,>=0.3.19; extra == "all"
dask: Missing, need dask[array]>=2.0.0; extra == "all"
drms: Missing, need drms>=0.6.1; extra == "all"
glymur: 0.9.9
h5netcdf: Missing, need h5netcdf>=0.8.1; extra == "all"
h5py: Missing, need h5py>=3.1.0; extra == "all"
matplotlib: 3.5.2
mpl-animators: Missing, need mpl-animators>=1.0.0; extra == "all"
pandas: Missing, need pandas>=1.0.0; extra == "all"
python-dateutil: 2.8.2
reproject: Missing, need reproject; extra == "all"
scikit-image: Missing, need scikit-image<0.19,>=0.16.0; extra == "all"
scipy: 1.8.0
sqlalchemy: Missing, need sqlalchemy>=1.3.4; extra == "all"
tqdm: 4.64.0
zeep: Missing, need zeep>=3.4.0; extra == "all"
```

### Installation method

pip

**Comment by nabobalis** [comment]
So it seems they are missing the `detector` keyword that is present in the FITS files. I can add a work around for this in the source file.

**Comment by dgarciabriseno** [comment]
Thanks, interestingly this works with an older version of sunpy (2.0.5). Was the previous version looking at different fields?

**Comment by nabobalis** [comment]
Yeah installing 3.0, I can open the JP2 files no issue, so it just a problem for 3.1, which I have fixed in #6112

**Comment by dgarciabriseno** [comment]
Thanks!
